### PR TITLE
mac:(launchd) mange the WS App as a service

### PR DESCRIPTION
**Pull request by afiune**
## Description
This change is adding a new `scripts/` folder with a shell script
called `chef_workstation_app_launcher` that will control the app
launcher behavior for MacOS systems.

Load the app as a service:
```
$ chef_workstation_app_launcher load
```

Remove the app as a service:
```
$ chef_workstation_app_launcher remove
```

Show launchd information about the app as a service:
```
$ chef_workstation_app_launcher show
{
	"LimitLoadToSessionType" = "Aqua";
	"MachServices" = {
		"sh.chef.chef-workstation.MachPortRendezvousServer.6956" = mach-port-object;
	};
	"Label" = "chef.workstation.app";
	"TimeOut" = 30;
	"OnDemand" = true;
	"LastExitStatus" = 0;
	"PID" = 6956;
	"Program" = "/Applications/Chef Workstation App.app/Contents/MacOS/Chef Workstation App";
	"ProgramArguments" = (
		"/Applications/Chef Workstation App.app/Contents/MacOS/Chef Workstation App";
	);
	"PerJobMachServices" = {
		"com.apple.tsm.portname" = mach-port-object;
		"com.apple.coredrag" = mach-port-object;
		"com.apple.axserver" = mach-port-object;
	};
};
```

### Usage
```
$ /opt/chef-workstation/bin/chef_workstation_app_launcher --help
Usage: chef_workstation_app_launcher <subcommand>

Controls the Chef Workstation App launcher behavior for MacOS systems.

Subcommands:
    show      Show launchd information about the Chef Workstation App service.
    load      Bootstraps the Chef Workstation App service.
    remove    Removes the Chef Workstation App service.
```

Signed-off-by: Salim Afiune <afiune@chef.io>

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Chore (non-breaking change that does not add functionality or fix an issue)

## Checklist:

- [x] I have read the **CONTRIBUTING** document.
- [ ] I have run the pre-merge tests locally and they pass.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.
- [x] All commits have been signed-off for [the Developer Certificate of Origin](https://github.com/chef/chef/blob/master/CONTRIBUTING.md#developer-certification-of-origin-dco).

### CODE REVIEW

**Comment by afiune** [comment]
Launched an [ad hoc build](https://buildkite.com/chef/chef-chef-workstation-master-omnibus-adhoc/builds/199) but for some reason the MacOS packages are not being
tagged correctly so I can't find the correct package, I am trying to build locally now.

**Comment by afiune** [comment]
With the commit https://github.com/chef/chef-workstation/pull/913/commits/163762199d410987355b129f2a660931db2d5e68 now the [ad hoc build is passing](https://buildkite.com/chef/chef-chef-workstation-master-omnibus-adhoc/builds/202)

cc/ @jonsmorrow 

tenor-138634906

**Comment by afiune** [comment]
@tyler-ball nope. I just made my own 😂 I did read the man page for the launchd daemon agent:

https://www.manpagez.com/man/5/launchd.plist/

**Comment by tyler-ball** [comment]
Even more impressive!

**Comment by afiune** [comment]
@tyler-ball I wanna disclose that, I had already thought about this launcher script
since I needed to control the "start at boot" via a checkbox inside the Preferences
Dialog (PR https://github.com/chef/chef-workstation-app/pull/125). Now, with this
script, we will be able in the future to control such behavior by adding an extra sub-
command called (maybe) `disable`.

Idea:
```
$ /opt/chef-workstation/bin/chef_workstation_app_launcher --help
Usage: chef_workstation_app_launcher <subcommand>

Controls the Chef Workstation App launcher behavior for MacOS systems.

Subcommands:
    show      Show launchd information about the Chef Workstation App service.
    load      Bootstraps the Chef Workstation App service.
    remove    Removes the Chef Workstation App service.
    disable   Disable the Chef Workstation App to start at boot.
```

File: assets/scripts/chef_workstation_app_launcher [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+PROGNAME=$(basename "$0")
+PRODUCTNAME="Chef Workstation App"
+SERVICENAME="io.chef.chef-workstation.app"
+PLISTFILE="${SERVICENAME}.plist"
+
+usage() {
+  cat <<DOC
+Usage: ${PROGNAME} <subcommand>
+
+Controls the ${PRODUCTNAME} launcher behavior for MacOS systems.
```
Comment (tas50) [inline assets/scripts/chef_workstation_app_launcher]:
```suggestion
Controls the ${PRODUCTNAME} launcher behavior for macOS systems.
```

File: assets/scripts/chef_workstation_app_launcher [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+    show      Show launchd information about the ${PRODUCTNAME} service.
+    load      Bootstraps the ${PRODUCTNAME} service.
+    remove    Removes the ${PRODUCTNAME} service.
+DOC
+}
+
+main()
+{
+  if ! is_darwin; then
+    error_exit "Launcher is only available for MacOS systems"
```
Comment (tas50) [inline assets/scripts/chef_workstation_app_launcher]:
```suggestion
    error_exit "Launcher is only available for macOS systems"
```

**Comment by afiune** [comment]
@jonsmorrow I will change it. My only concern is that users might get confused by the name today since this is very specific for the WS App. Hopefully not 🤞

**Comment by jonsmorrow** [comment]
@afiune we could add to the help out put a list of supported options or have it launch all without an argument

**Comment by afiune** [comment]
@jonsmorrow it already does that 💯

**Pull request merged**